pkg/cmd/server/api/types.go

```diff
@@ -40,6 +49,10 @@ type NodeConfig struct {
 	// PodManifestConfig holds the configuration for enabling the Kubelet to
 	// create pods based from a manifest file(s) placed locally on the node
 	PodManifestConfig *PodManifestConfig
+
+	// DockerExecHandlerName is the name of the handler to use for executing
+	// commands in Docker containers.
+	DockerExecHandlerName DockerExecHandlerType
```

Comment:
> Conversions!
> @deads2k ???

Sorry, I got overly excited at the prospect of carrying conversion functions to move `AllowDisabledDocker` into a `DockerOptions` stanza of our node config.

Let's start `DockerOptions` with the `DockerExecHandlerName` and deal with refactoring other pieces later.